src/components/TextInput/TextInputTypes.tsx

```diff
@@ -0,0 +1,38 @@
+// HTML Input types as defined by MDN: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input
+export enum TextInputTypes {
+  email = "email",
+  hidden = "hidden",
+  // eslint-disable-next-line id-blacklist
+  number = "number",
+  password = "password",
+  text = "text",
+  textarea = "textarea",
+  tel = "tel",
+  url = "url",
+}
+
+export enum TextInputFormats {
+  email = "jdoe@domain.com",
+  hidden = "",
+  // eslint-disable-next-line id-blacklist
+  number = "",
+  password = "",
+  text = "",
+  tel = "(123) 123-1234",
+  textarea = "",
+  url = "https://domain.com",
+}
+
+export enum TextInputPatterns {
+  // eslint-disable-next-line prettier/prettier
+  email = "[a-z0-9._%+-]+@[a-z0-9.-]+.[a-z]{2,}$",
+  hidden = "",
+  // eslint-disable-next-line id-blacklist
+  number = "",
+  password = "",
+  text = "",
+  // eslint-disable-next-line prettier/prettier
+  tel = "([0-9]{3}) [0-9]{3}-[0-9]{4}",
+  textarea = "textarea",
+  url = "https?://.+",
+}
```

Comment:
Do these get used anywhere?  On the subject, I wonder if we should offer the user the choice of enabling HTML5 field validation or not.  If so, we could also allow them to enter a custom pattern or select from the defaults we have in the enum above.

Reply:
@kaplanmichael 
Those are leftover from my development process.  I was planning on using them, but React was stripping out some of the formatting in the string.  At the momenet, I don't recall the exact problem, but it was related to the backslash character.

HTML5 validation is activated automatically based on the type attribute.  The `pattern` value enhances that validation.

Right now, `pattern` can be passed in as part of the attributes.  That is currently being done on the Input Attribiutes story.

Reply:
This can be done by adding `novalidate` for a `<form>` tag.